packages/relic/lib/src/headers/headers.dart

```diff
@@ -0,0 +1,1482 @@
+import 'dart:io' as io;
+
+import 'package:http_parser/http_parser.dart';
+import 'package:relic/src/headers/custom/custom_headers.dart';
+import 'package:relic/src/headers/extension/string_list_extensions.dart';
+import 'package:relic/src/headers/parser/headers_parser.dart';
+import 'package:relic/src/headers/parser/shared_parser.dart';
+import 'package:relic/src/method/method.dart';
+
+import 'typed/typed_headers.dart';
+
+import '../body/body.dart';
+
+abstract class Headers {
```

Comment:
If you don't make this a `final class` then people outside this package can implement / extend it, and you'll have problems if you add new headers to it in the future.

It could potentially be a breaking change to add a new field to a class that isn't `final` or `sealed`.

Reply:
@jonasfj,  
I agree with you, but if someone implements this class, they must override and implement all abstract methods. If those methods are implemented incorrectly, it's beyond our responsibility.  

That said, we should review and decide if making it a `final` class is better to prevent unintended implementations.